packages/devtools_app/lib/src/debugger/flutter/debugger_screen.dart

```diff
@@ -338,3 +344,78 @@ class DebuggingControls extends StatelessWidget {
     );
   }
 }
+
+class DebuggerStatus extends StatefulWidget {
+  const DebuggerStatus({
+    Key key,
+    @required this.controller,
+  }) : super(key: key);
+
+  final DebuggerController controller;
+
+  @override
+  _DebuggerStatusState createState() => _DebuggerStatusState();
+}
+
+class _DebuggerStatusState extends State<DebuggerStatus> with AutoDisposeMixin {
+  String _status;
+
+  @override
+  void initState() {
+    super.initState();
+
+    addAutoDisposeListener(widget.controller.isPaused, _updateStatus);
+
+    _status = '';
+    _updateStatus();
+  }
+
+  @override
+  void didUpdateWidget(DebuggerStatus oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    // todo: should we check that widget.controller != oldWidget.controller?
+    addAutoDisposeListener(widget.controller.isPaused, _updateStatus);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Text(
+      _status,
+      maxLines: 1,
+      overflow: TextOverflow.ellipsis,
+    );
+  }
+
+  void _updateStatus() async {
+    final paused = widget.controller.isPaused.value;
+
+    if (!paused) {
```

Comment:
nit: consider adding a 
`String computeStatus()`
method to remove all the pairs of `setState and return`
e.g.
```dart
final status = computeStatus();
if (status != _status) {
  setState(() {
    _status = status;
  });
}
```

Reply:
sounds good, will do